src/ApplicationInsights.Kubernetes/Extensions/AppInsightsForKubernetesOptions.cs

```diff
@@ -21,5 +21,11 @@ public class AppInsightsForKubernetesOptions
         /// </summary>
         [JsonProperty("DisablePerformanceCounters")]
         public bool DisablePerformanceCounters { get; set; }
+
+        /// <summary>
+        /// Gets or sets the processor for telemetry key. This is introduced to allow customization of
+        /// telemetry keys.
+        /// </summary>
+        public Func<string, string> TelemetryKeyProcessor { get; set; }
```

Comment:
Alternatively this could be a setting, or a couple of settings: a search pattern, and a replacement string, using either [ordinary string replacement](https://docs.microsoft.com/en-us/dotnet/api/system.string.replace?view=net-5.0#System_String_Replace_System_String_System_String_System_StringComparison_) or [regex replace](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.replace?view=net-5.0#System_Text_RegularExpressions_Regex_Replace_System_String_System_String_System_String_System_Text_RegularExpressions_RegexOptions_System_TimeSpan_). 

Regex can be costly, so I would do it as opt-in, but it really depends on the specific regular expression, and we would probably compile the regexp for fast execution.

Reply:
Also, this contract does not prevent the telemetry key processor to return **different** transformed key for the same input (upon every invocation). Maybe we consider this a "misbehaving" processor, but we should probably document this, otherwise the caching of the transform result is not possible.

Reply:
This is a great question. It is tough to balance functionality and security. Being hit off guard by this requirement makes me want to provide a way to allow address all cases. For example, the user could prefer to hash the key rather than simple replacement; That said, I see your concerns on ever growing transformed-key list.

I think I'll `document it` + `limit the cap of the cache entry number` for now.